circuitpython/code.py

```diff
@@ -114,23 +114,32 @@ def message(client, feed_id, payload):
 
 # Below is an example of manually publishing a new  value to Adafruit IO.
 last = 0
-print("Publishing a new message every 5 seconds...")
+print("Publishing a new message every 20 seconds...")
 while True:
     # Explicitly pump the message loop.
-
+    
     # Send a new message every 5 seconds.
     if (time.monotonic() - last) >= 20:
-        input_value_1 = get_voltage(input_voltage_1)
-        input_value_2 = get_voltage(input_voltage_2)
+        input_value_1 = get_voltage(input_voltage_1) * 3 # the current over 0.5 Ohm as voltage
+        # offset correction: the measured solar voltage includes the voltage drop over the 
+        # 0.5 Ohm current resistor. But now we've taken this measurement and can subtract
+        # it to get the correct value: input_voltage_1 is now the voltage over a 0.5 Ohm
+        # resistor times 3, so we could subtract this value divided by 3 from the solar voltage
+        # the voltage we measure is the solar voltage divided by 5.7 (47 kOhm and 10 kOhm series)
+        # so we have to divide the input_value_1 by 3 and 5.7 before subtracting from
+        # the submitted value for input_value_2
+        
+        input_value_2 = get_voltage(input_voltage_2) - (input_value_1 / (3 * 5.7))
+        # the solar voltage divided by 5.7
```

Comment:
Why not divided by 17.1? Change voltage divider to new position?

And why not move the voltage divider to the negative of the solar panel?